collector/item_analysis.go

```diff
@@ -0,0 +1,310 @@
+package main
+
+import (
+	"context"
+	"crypto/sha256"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/avast/retry-go"
+	"github.com/gorilla/websocket"
+	"github.com/mitchellh/mapstructure"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/corshatech/cast/collector/analysis/pass_in_url"
+)
+
+type workerContext struct {
+	Ctx          context.Context
+	KsConnection *websocket.Conn
+	KsHubUrl     string
+	PgConnection *sql.DB
+	Queue        <-chan Message
+	WorkerNo     int
+}
+
+func createAnalysisPool(pgConnection *sql.DB, ksConnection *websocket.Conn, kubesharkHubURL string, workerBufSize, workerNo int, ctx context.Context) error {
+	messageQueue := make(chan Message, workerBufSize)
+
+	err := ksConnection.SetWriteDeadline(deadline(defaultWriteDeadline))
+	if err != nil {
+		log.WithError(err).Error("Unable to set connection deadline")
+		return err
+	}
+	err = ksConnection.WriteMessage(websocket.TextMessage, []byte{})
+	if err != nil {
+		log.Println("write:", err)
+		return err
+	}
+
+	for i := 0; i < workerNo; i++ {
+		workCtx := workerContext{
+			Ctx:          ctx,
+			KsConnection: ksConnection,
+			KsHubUrl:     kubesharkHubURL,
+			PgConnection: pgConnection,
+			Queue:        messageQueue,
+			WorkerNo:     i,
+		}
+
+		go analysisWorker(&workCtx)
+	}
+
+	log.Info("Starting export of records.")
+
+	for {
+		err = ksConnection.SetReadDeadline(deadline(defaultReadDeadline))
+		if err != nil {
+			log.WithError(err).Error("Unable to set connection deadline")
+			return err
+		}
+		_, messageJson, err := ksConnection.ReadMessage()
+		if err != nil {
+			log.Println("read:", err)
+			return err
+		}
+
+		message := Message{}
+		err = json.Unmarshal(messageJson, &message)
+		if err != nil {
+			log.WithError(err).Error("Failed to unmarshal websocket message")
+			return fmt.Errorf("failed to unmarshal websocket message: %w", err)
+		}
+
+		if message.Proto.Name != "http" {
+			log.WithField("proto.name", message.Proto.Name).Debug("ignoring non-HTTP traffic")
+			return nil
+		}
+
+		select {
+		case messageQueue <- message:
+			// Nothing
+		default:
+			log.Warn("Can't keep up! Traffic was discarded.")
+		}
+
+		select {
+		case <-ctx.Done():
+			close(messageQueue)
+			return nil
+		default:
+			continue
+		}
+	}
+}
+
+func analysisWorker(ctx *workerContext) {
+	for message := range ctx.Queue {
+		itemUrl := fmt.Sprintf("%s/item/%s?q=", ctx.KsHubUrl, message.Id)
+		requestContext, cancel := context.WithDeadline(ctx.Ctx, deadline(1*time.Minute))
```

Comment:
It doesn't make sense to cache a context object like this, since this is a standalone goroutine it's fine to base this context off of context.Background()
```suggestion
		requestContext, cancel := context.WithDeadline(context.Background(), deadline(1*time.Minute))
```